src/screens/Assistant/ResultItem.tsx

```diff
@@ -113,6 +118,15 @@ const ResultItemHeader: React.FC<{
         </AccessibleText>
       </View>
 
+      {tripIncludesRailReplacementBus && (
+        <Warning
+          accessibilityLabel={t(
+            AssistantTexts.results.resultItem.tripIncludesRailReplacementBus,
+          )}
+          style={styles.warningIcon}
+        />
+      )}
```

Comment:
Things like this could also be it's own component where the logic is encapsulated. Naming it something like `WarnWhenRailReplacementBus` or something.

Reply:
I don't think it necessarily makes things less complex, but at least it makes the return function cleaner. Did not extract it into a separate file or something, as the `Assistant` version is to be removed some day, and the texts are fetched from two different text files. 

Was this in line with what you had in mind, @mikaelbr?

Reply:
Does it need to have separate texts? My point is, _is_ it the same or does it just look the same? Translated texts are an abstraction and not semantically difference. If it _is_ the same, you can have a component shared across and have that component it's own text file.

Reply:
I've extracted the encapsulated logic where the texts actually are the same, and put it into a separate component structure. Starting to feel a bit like over-engineering something quite simple, but the code looks cleaner in the original file at least.